streamtools/producers.py

```diff
@@ -207,6 +207,9 @@ def add_agent_uuid(self, **kwargs_from_agent):
         self.routing_key += f"-{self.key[:5]}"
 
     async def send(self, msg, *args, **kwargs):
+        if self.channel.is_closed:
```

Comment:
Do we want this to also happen on the read side? Or will it auto-open the channel if needed to read?

Reply:
I'm thinking of this as more of a really targeted temporary fix for now to exactly target where the problem is popping up.

Ideally, I'd restructure the code to use a context manager or something for a more robust fix across the `streamtools` codebase (to be tackled in CP-1667)